WordPress/Classes/Services/PostService.m

```diff
@@ -526,8 +526,19 @@ - (void)updatePost:(AbstractPost *)post withRemotePost:(RemotePost *)remotePost
     post.permaLink = [remotePost.URL absoluteString];
     post.content = remotePost.content;
     post.status = remotePost.status;
-    post.password = remotePost.password;
-    post.post_thumbnail = remotePost.postThumbnailID;
+    post.password = remotePost.password;    
+    if (remotePost.postThumbnailID != nil) {
+        Media *featuredMedia = [Media existingMediaWithMediaID: remotePost.postThumbnailID inBlog:post.blog];
```

Comment:
We could extract this into a helper method on `Media` – `existingOrStubMediaWithMediaID:inBlog:`